Hosts cli

azjones227 opened the pull request:
#3 feature added. at current only the flag --add is necessary. Refactored hostcontrol.js to properly use try/catch with await functions.

azjones227 committed:
#3 make branch hostsCLI current with local working directory

azjones227 committed:
#3 feature added, code refactor to properly use try/catch with await functions

wmantly reviewed (changes requested):
What if the user doesn't put a hostname or IP? what if they put one? How does a user know what to put? you should build `--help` as you go.

wmantly commented on hostcontrol.js:
```diff
@@ -7,16 +7,32 @@ const command = process.argv[2];
 		process.exit(0)
+		} catch(e){
+			console.error(e)
+			process.exit(2)
```
What does exit code `2` mean?

azjones227 replied:
Exit code 2 is for when the CLI does not complete its intended task. Prior to that the underlying hosts.js function should report what the error is. This has not been tested yet.

azjones227 replied:
After reviewing http://tldp.org/LDP/abs/html/exitcodes.html, i changed the exit code 2 to exit code 1. If the flag --help is used the script will exit with code 0, if theres a typo in the command the script will exit with code 127.

wmantly commented:
In the help section, write a map for each error code, like;
```
Error code map
0 -- ok
1 -- Missing user input
2 -- internal error
3 -- command not found
```

wmantly commented:
o snap, there are some standards with error codes
http://tldp.org/LDP/abs/html/exitcodes.html

azjones227 committed:
#8 feature added. Rough draft of help text added. Error codes corrected.

azjones227 commented:
CLI should be refactored: 
1) use a dict where the key is the <--flag> and the value is the function
2) concept of the CLI controller and view

azjones227 committed:
Changed hostcontrol.js >> hostctrl.js, added --remove

azjones227 committed:
Delete hostcontrol.js

azjones227 committed:
Delete hostctrl.js

File structure is changed. The files related to the CLI are now found in the cli directory.

azjones227 committed:
file structure is changed. Bring repo current with local files.

azjones227 committed:
file structure is changed. Bring repo current with local files.

azjones227 committed:
sync repo with local directory

azjones227 committed:
cli flag --ls-f added

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+			await views.list(await hosts.listAll())
+			process.exit(0)
+		} catch(e){
+			console.error(e)
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+			process.exit(0)
+
+		} catch(e){
+			console.error(e)
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+			console.log(data.host + " has been added successfully")
+			process.exit(0)
+		} catch (e){
+			console.error(e)
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+			views.info(data, line)
+			process.exit(0)
+		}catch(e){
+			console.error(e)
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+			console.log(data.host + " has been removed successfully")
+			process.exit(0)
+		}catch(e){
+			console.error(e)
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+		process.exit(0)
+
+	} else {
+		console.error("Command not found. Use --help for options")
```
An error view would be better.

wmantly commented on cli/controller.js:
```diff
@@ -0,0 +1,82 @@
+//set command variable
+const command = process.argv[2];
+
+(async function(command){
```
Ths async IEFE belongs in `bin/cli` and the controller should export all the functionality.
